packages/beacon-node/src/eth1/eth1MergeBlockTracker.ts

```diff
@@ -115,232 +163,175 @@ export class Eth1MergeBlockTracker {
+    if (this.status.code !== StatusCode.MERGE_COMPLETE) {
+      this.status = {code: StatusCode.FOUND, mergeBlock};
+      this.metrics?.eth1.eth1MergeBlockDetails.set(
+        {
+          terminalBlockHash: mergeBlock.blockHash,
+          // Convert all number/bigints to string labels
+          terminalBlockNumber: mergeBlock.number.toString(10),
+          terminalBlockTD: mergeBlock.totalDifficulty.toString(10),
+        },
+        1
+      );
     }
+  }
 
-    // TTD already reached, search blocks backwards
-    let minFetchedBlockNumber = latestBlock.number;
-    // eslint-disable-next-line no-constant-condition
-    while (true) {
-      const from = Math.max(0, minFetchedBlockNumber - MAX_BLOCKS_PER_PAST_REQUEST);
-      // Re-fetch same block to have the full chain of parent-child nodes
-      const to = minFetchedBlockNumber;
-
-      try {
-        const blocksRaw = await this.eth1Provider.getBlocksByNumber(from, to);
-        const blocks = blocksRaw.map(toPowBlock);
-
-        // Should never happen
-        if (blocks.length < 2) {
-          throw Error(`getBlocksByNumber(${from}, ${to}) returned less than 2 results`);
-        }
-
-        for (let i = 0; i < blocks.length - 1; i++) {
-          const childBlock = blocks[i + 1];
-          const parentBlock = blocks[i];
-          if (
-            childBlock.totalDifficulty >= this.config.TERMINAL_TOTAL_DIFFICULTY &&
-            parentBlock.totalDifficulty < this.config.TERMINAL_TOTAL_DIFFICULTY
-          ) {
-            // Is terminal total difficulty block
-            if (childBlock.parentHash === parentBlock.blockHash) {
-              // AND has verified block -> parent relationship
-              return this.setTerminalPowBlock(childBlock);
-            } else {
-              // WARNING! Re-org while doing getBlocksByNumber() call. Ensure that this block is the merge block
-              // and not some of its parents.
-              return await this.fetchPotentialMergeBlock(childBlock);
-            }
-          }
-        }
-
-        // On next round
-        minFetchedBlockNumber = Math.min(to, ...blocks.map((block) => block.number));
-
-        // Scanned the entire blockchain
-        if (minFetchedBlockNumber <= 0) {
-          return;
-        }
-      } catch (e) {
-        if (!isErrorAborted(e)) this.logger.error("Error on fetchPreviousBlocks range", {from, to}, e as Error);
-        await sleep(SLEEP_ON_ERROR_MS, this.signal);
+  private async pollTerminalPowBlockFromEth1(): Promise<void> {
+    try {
+      const mergeBlock = await this.getTerminalPowBlockFromEth1();
+      if (mergeBlock) {
+        this.logger.info("Terminal POW block found!", {
+          hash: mergeBlock.blockHash,
+          number: mergeBlock.number,
+          totalDifficulty: mergeBlock.totalDifficulty,
+        });
+      }
+      // It is possible for status to be updated to merge complete or found or stopped
+      if (this.status.code != StatusCode.SEARCHING) {
```

Comment:
This condition doesn't look right, should it be this?
```suggestion
      if (this.status.code === StatusCode.SEARCHING) {
```

Reply:
no we want to clear intervals when its not searching, since this codeflow is called by the poll